core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3394,6 +3399,10 @@ protected function reviseTable()
 			{
 				$objIds = $this->Database->execute("SELECT c.id FROM " . $this->strTable . " c LEFT JOIN " . $ptable . " p ON c.pid=p.id WHERE c.ptable='" . $ptable . "' AND p.id IS NULL");
 			}
+			elseif ($ptable == $this->strTable)
+			{
+				$objIds = $this->Database->execute('SELECT c.id FROM ' . $this->strTable . ' c LEFT JOIN ' . $ptable . ' p ON c.pid=p.id WHERE p.id IS NULL AND c.pid > 0');
+			}
 			else
 			{
 				$objIds = $this->Database->execute("SELECT c.id FROM " . $this->strTable . " c LEFT JOIN " . $ptable . " p ON c.pid=p.id WHERE p.id IS NULL");
```

Comment:
Can’t we always check for `c.pid > 0` regardless of the mode?
```suggestion
			else
			{
				$objIds = $this->Database->execute('SELECT c.id FROM ' . $this->strTable . ' c LEFT JOIN ' . $ptable . ' p ON c.pid=p.id WHERE p.id IS NULL AND c.pid > 0');
```

Reply:
no, if a `tl_content` record has pid=0 it should be deleted

Reply:
Why? The core never creates `tl_content` rows with `pid=0`, or does it?

Reply:
Probably not, but incorrectly imported data for example could lead to these invalid records. `reviseTable` should clean up everything that is incorrect, not just our core cases.